### ISSUE TITLE
Switch to PhysFS

### ISSUE THREAD

**Issue by qwertychouskie** [cubosphere/cubosphere-code#31]
Poco is currently disabled for Windows builds in my local working copy due to me having infinite headaches trying to get it to build for Windows and be included properly into Cubosphere.  Also, Poco is very large, providing a lot of functionality not needed.

Enter PhysFS.  (https://icculus.org/physfs/)  From the description, it sounds like exactly what we need, and if we use it we would likely be able to get rid of most of `filesystem.cpp`, likely leaving only a few wrappers and helpers for the Lua scripts.

PhysFS seems to be kept reasonably up-to-date (there was a bugfix release in March of this year), and is used by many programs, including SuperTux.

What are your thoughts?

**Comment by v1993** [comment]
@qwertychouskie Yeah, PhysFS is great library, I really like it. When working on from-scratch version I'll ceternaly use it.

If you'll replace it by yourself, I'll merge it. If not, that's ok.
